### PR TITLE
Draft: level@6

### PR DESCRIPTION
**Pull request by mixmix**
Wasn't sure how to develop on this repo (I'm new to lerna, jest, ts). 

Install?
```bash
npm i
cd packages/leveldb
npm i
cd ../..
```

The loop I was running:
```bash
npm run build
npm test
```

---

TODO
- [x] install `level@6`
- [ ] get tests passing
    - code gets stuck D:

### CODE REVIEW

File: packages/leveldb/src/leveldb/internal.ts [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import { RxDocumentData, RxJsonSchema, getPrimaryFieldOfPrimaryKey } from "rxdb";
-import { Level } from 'level';
+import Level from 'level';
+import pull from 'pull-stream';
+import pullLevel from 'pull-level';
```
Comment (mixmix) [inline packages/leveldb/src/leveldb/internal.ts:4]: `pull-stream` is an elegant async-iterator library that Scuttlebutt uses extensively.

File: packages/leveldb/src/leveldb/internal.ts [hunk 2]
```diff
@@ -36,20 +40,26 @@ export class LevelDBInternal<RxDocType> implements LevelDBStorageInternals<RxDoc
...
+              pullLevel.read(db),
+              pull.filter(row => !isArray(row.value)),
+              pull.map(row => {
+                docsInDbMap.set(row.key, row.value)
+                return
+                // don't return anything - we don't care about collecting
+              }),
+              pull.collectAsPromise()
+            )
+            // WIP - not reaching here?
```
Comment (mixmix) [inline packages/leveldb/src/leveldb/internal.ts:62]: We're never reaching this point in the tests. Unknown why

File: packages/leveldb/test.mjs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+await db.put('cat', 'nala')
+
+await pull(
+  pullLevel.read(db),
+  pull.map(({ key, value }) => {
+    console.log(key, '=', value)
+    return
+  }),
+  pull.collectAsPromise()
+)
```
Comment (mixmix) [inline packages/leveldb/test.mjs:19]: this comparable code runs perfectly

**Comment by elribonazo** [comment]
Very nice! I'm literally checking this now from a plane. You made my day man!

**Comment by elribonazo** [comment]
u should just npm i and npm run build or npm run tests if i'm not wrong.

File: packages/leveldb/test.mjs (hunk 3 above)
Comment (elribonazo) [reply packages/leveldb/test.mjs:19]: can u try running this in 1 subthread or something like this?

obviously 2 threads will not work but what i want to see is, if the solution is just replacing jest by mocha or whatever.
I appreciate this changes a lot man, thank you!

**Comment by elribonazo** [comment]
@mixmix still flying lol so invested time on checking this properly, hmm, the error we are having is the same:
Error: IO error: lock prism-db/LOCK: already held by process.

Lets continue taking a look, I's probably just find another testing suite that doesn't give soo mich trouble with single threading. I did, there's an option -i, to make sure it runs on 1 machine, despite that when i open the debugger seems to be more than 1 instance.

Curiosuly also in the index for levelDB, we try not recharging the instance eaach time its called so mostly sure its subthreads or another Jest process even.

As soon as we fix JEst, or the test engine, then we will have leveldb7,8, and 1006 :D all of them. Have moves your code to this branch

**Comment by elribonazo** [comment]
Also made you a maintainer so u don't need to fork it, i find it a little hard like this :). as we are not many here hehe

**Comment by elribonazo** [comment]
I'd say level 8.0.0 is more what we need as its typesafe. The libraries we are using are not typesafe and a little bit risky. 

I'll continue fixing 8.0.0 but i don't think honestly we should have a lower version. Its hard to maintain, edit and understand, not good :) Latest versions are not perfect but better ..

**Comment by mixmix** [comment]
The problem with Level 8 is we don't have binaries that are yet guarenteed to work on mobile yet I think. Level relies on a native-binding and I would much rather user pre-built binaries than get into the hole of getting all that working. It's like fires on every front sometimes!

**Comment by mixmix** [comment]
Moved branch into this repo, Opened new tracking PR - https://github.com/elribonazo/pluto-encrypted/pull/40

**Pull request closed without merging**